x/shared/types/params.go

```diff
@@ -186,6 +191,18 @@ func validateClaimWindowOpenOffsetIsAtLeastGracePeriodEndOffset(params *Params)
 			params.GracePeriodEndOffsetBlocks,
 		)
 	}
+	return nil
+}
 
+// validateGracePeriodOffsetBlocksIsLessThanNumBlocksPerSession validates that the
+// GracePeriodEndOffsetBlocks is less than NumBlocksPerSession; i.e., one session.
+func validateGracePeriodOffsetBlocksIsLessThanNumBlocksPerSession(params *Params) error {
```

Comment:
As the grace period approaches the session duration, I think the session looses its meaning. It doesn't make sense to me that it would be valid for a gateway to be able to continue using a previous sessions suppliers for the entire duration of a subsequent session. This could create surface area for a denial of service attack vector for gateways against suppliers.